stable/database/templates/_helpers.tpl

```diff
@@ -221,3 +221,13 @@ The configuration is imported only in the entrypoint cluster.
 {{- end -}}
 {{- end -}}
 {{- end -}}
+
+{{- define "autoRestore.type" -}}
+{{- $valid := list "stream" "backupset" "" }}
+{{- if not (has .Values.database.autoRestore.type $valid) }}
+{{- fail "Invalid autorestore type" }}
+{{- end }}
+{{- with .Values.database.autoRestore.type }}
+{{- . }}
```

Comment:
Should we keep the `default "stream" .Values.database.autoRestore.type`? Otherwise empty value will not result in `stream` as before.

Reply:
```
$ helm install --dry-run db stable/database --set database.autoRestore.type=garbage
Error: template: database/templates/configmap.yaml:73:30: executing "database/templates/configmap.yaml" at <include "autoRestore.type" .>: error calling include: template: database/templates/_helpers.tpl:228:4: executing "autoRestore.type" at <fail "Invalid autorestore type">: error calling fail: Invalid autorestore type
```

Its pretty cool! It fails your Helm installation straight away, rather than fail at runtime.